PurchasesCoreSwift/Misc/ISOPeriodFormatter.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  ISOPeriodFormatter.swift
+//  PurchasesCoreSwift
+//
+//  Created by Joshua Liebowitz on 6/30/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+import StoreKit
+
+@available(iOS 11.2, macOS 10.13.2, tvOS 11.2, *)
+
+// TODO(post-migration): Switch public back to <> (nothing)
+// TODO(post-migration): remove @objc since this is internal only
+@objc(RCISOPeriodFormatter) public class ISOPeriodFormatter: NSObject {
+
+    @objc public func stringFromProductSubscriptionPeriod(_ period: SKProductSubscriptionPeriod) -> String {
+        let unitString = self.period(fromUnit: period.unit)
+        let stringResult = "P\(period.numberOfUnits)\(unitString)"
+        return stringResult
+    }
+
+    private func period(fromUnit unit: SKProduct.PeriodUnit) -> String {
+        switch unit {
+        case .day:
+            return "D"
+        case .week:
+            return "W"
+        case .month:
+            return "M"
+        case .year:
+            return "Y"
+        @unknown default:
+            fatalError()
```

Comment:
I highly doubt this would ever come up, but: should we have a message or something to indicate what's happening? 
realistically I think the benefit of `@unknown default` here is to prevent the xcode warning, but unless the world switches calendar formats and introduces a new entity besides days, weeks, months and years we're fine 😂

Reply:
I agree, I added some text to the fatalerror